### PR TITLE
Docs: Burrow Integration Fixes

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Fixes on tasks https://jira.hyperledger.org/browse/IR-852, https://jira.hyperledger.org/browse/IR-842, https://jira.hyperledger.org/browse/IR-843
+minor fixes

Also added new tools for Iroha, suggested by the community!

### Benefits

Added info on permissions, fixed some errors. Added new information

### Possible Drawbacks 

Something might still be wrong. This PR needs a very thorough review

### CODE REVIEW

**Comment by baziorek** [comment]
I've checked and it looks OK:)

File: docs/source/develop/api/commands.rst [hunk 1]
```diff
@@ -236,7 +236,7 @@ Validation
 ^^^^^^^^^^
 
 1. Caller is a valid Iroha account ID
-2. The transaction creator has a role with either CanCallEngine permission
+2. The transaction creator has a role with can_call_engine permission
```
Comment (MBoldyrev) [inline docs/source/develop/api/commands.rst:239]: maybe the logically missing part was supposed to address `can_call_engine_on_my_behalf`, which we should add here

File: docs/source/permissions/matrix.csv [hunk 2]
```diff
@@ -29,17 +29,19 @@ Command,Grant,can_grant_can_transfer_my_assets,FALSE,,Allows role owners grant `
...
-Query,Account,can_get_all_acc_detail,FALSE,,Allows getting all the details set to any account within the system.,,,../api/queries.html#get-account-detail,Admin creates Alice account in a diffrerent domain that has only can_get_all_acc_detail permission. Alice can access details set to Admin account.
+Command,Engine,can_call_engine,FALSE,,Allows to use Burrow EMV to run Solidity smart-contracts,,,../api/commands.html#call-engine,Admin creates domain that contains only can_call_engine permission and Alice account in that domain. Alice can send Solidity smart contracts to Burrow EVM by using Call Engine command.
+Command,Grant,can_grant_can_call_engine_on_my_behalf,FALSE,,Allows role owners grant `can_call_engine`  permission.,,,../api/commands.html#call-engine,Admin creates domain that contains only can_grant_can_call_engine permission and two accounts for Alice and Bob in that domain. Alice can grant to Bob and revoke can_call_engine permission.
```
Comment (MBoldyrev) [inline docs/source/permissions/matrix.csv:43]:
```suggestion
Command,Grant,can_grant_can_call_engine_on_my_behalf,FALSE,,Allows role owners grant `can_call_engine_on_my_behalf`  permission.,,,../api/commands.html#call-engine,Admin creates domain that contains only can_grant_can_call_engine permission and two accounts for Alice and Bob in that domain. Alice can grant to Bob and revoke `can_call_engine_on_my_behalf` permission.
```

Comment (MBoldyrev) [reply docs/source/permissions/matrix.csv:43]: maybe there should also be an entry for `can_call_engine_on_my_behalf`

**Comment by MBoldyrev** [comment]
also I think we need it in support/1.2

**Pull request merged**